jenkins/release-commit-lock/release-commit-lock.jenkinsfile

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+lib = library(identifier: 'jenkins@6.4.0', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/opensearch-project/opensearch-build-libraries.git',
+]))
+
+pipeline {
+    options {
+            timeout(time: 2, unit: 'HOURS')
+    }
+    agent {
+        docker {
+            label 'Jenkins-Agent-AL2023-X64-C54xlarge-Docker-Host'
+            image 'opensearchstaging/ci-runner:ci-runner-centos7-opensearch-build-v3'
+            registryUrl 'https://public.ecr.aws/'
+            alwaysPull true
+        }
+    }
+    parameters {
+        string(
+            name: 'RELEASE_VERSION',
+            description: 'The Release Version.',
+            trim: true
+        )
+        string(
+            name: 'OPENSEARCH_RELEASE_CANDIDATE',
+            description: 'The OpenSearch Release Candidate build number.',
+            trim: true
+        )
+        string(
+            name: 'OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE',
+            description: 'The OpenSearch Dashboards Release Candidate build number.',
+            trim: true
+        )
+        choice(
+                choices: ['Match Build Manifest', 'Update to Recent Commits'],
+                name: 'MANIFEST_LOCK_ACTION',
+                description: 'The manifest lock action to choose.<br>Match Build Manifest: Will update the manifest with commit ID from release candidate build manifest.<br>Update to Recent Commits: Will update the manifest with component repo release branch head commit.',
+        )
+        string(
+            name: 'COMPONENTS',
+            description: 'The selected components from the manifesr to updatet the commit ID',
+            trim: true
+        )
+    }
+    stages {
+        stage('Parameters Check') {
+            steps {
+                script {
+                    currentBuild.description = """Action: ${MANIFEST_LOCK_ACTION}<br>Release: ${RELEASE_VERSION} OS=${OPENSEARCH_RELEASE_CANDIDATE} OSD=${OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE}"""
+                    if (MANIFEST_LOCK_ACTION.isEmpty() || RELEASE_VERSION.isEmpty() || OPENSEARCH_RELEASE_CANDIDATE.isEmpty() || OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE.isEmpty()) {
+                                currentBuild.result = 'ABORTED'
+                                error('Make sure all the parameters are passed in.')
+                    }
+                }
+            }
+        }
+        stage('Match Build Manifest') {
+            when {
+                expression { params.MANIFEST_LOCK_ACTION == 'Match Build Manifest' }
+            }
+            steps {
+                script {
+                    def updateManifest = { String productName, String releaseCandidate ->
```

Comment:
Can we move this into a standalone method/function and can be called with different parameters? To reduce duplications?

Reply:
Notice @peterzhuamazon the variable `updateManifest` is the same but for each manifest lock action the functionality is different hence it is part of each stage, where the `def updateManifest` is unique for each script scope.